nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/compute_path_through_poses_action.hpp

```diff
@@ -68,7 +68,7 @@ class ComputePathThroughPosesAction
           "Destinations to plan through"),
         BT::InputPort<geometry_msgs::msg::PoseStamped>(
           "start", "Start pose of the path if overriding current robot pose"),
-        BT::InputPort<std::string>("planner_id", ""),
+        BT::InputPort<std::string>("planner_id", "Mapped name to the planner plugin type to use"),
```

Comment:
The entry string is the default, you would have to do `"", "description"` so that you don't set the default to the description for the string type. I"m not sure how else to get around that since the 2nd field could be cast into a string to be interpreted as a default

Reply:
You are right, atleast from the [templates here in this API](https://github.com/BehaviorTree/BehaviorTree.CPP/blob/725eba7e0abbc704284dec393706e5404f1472e3/include/behaviortree_cpp_v3/basic_types.h), I do not see any other option.

Reply:
But Input ports has just 2 templates, according to [here](https://github.com/BehaviorTree/BehaviorTree.CPP/blob/725eba7e0abbc704284dec393706e5404f1472e3/include/behaviortree_cpp_v3/basic_types.h):

You either have this:

https://github.com/BehaviorTree/BehaviorTree.CPP/blob/725eba7e0abbc704284dec393706e5404f1472e3/include/behaviortree_cpp_v3/basic_types.h#L282-L286 

or this:

https://github.com/BehaviorTree/BehaviorTree.CPP/blob/725eba7e0abbc704284dec393706e5404f1472e3/include/behaviortree_cpp_v3/basic_types.h#L300-L306

Reply:
Yes, but the `default` and `description` are both string types as the `dataType` is a `string` as well. I think you can see how that could be an issue that its impossible for the system to distinguish between the default value and the description for the case of string-based ports